From 0c2a8e755bb012ac4f2571faa95a4b5811a1cb36 Mon Sep 17 00:00:00 2001
From: Greg Rose <gvrose8192@gmail.com>
Date: Thu, 30 Jan 2020 15:55:27 -0800
Subject: [PATCH 01/19] compat: Remove HAVE_BOOL_TYPE

OVS only supports Linux kernels since 3.10 and all kernels since
then have the bool type.  This check is unnecessary so remove it.

Passes Travis:
https://travis-ci.org/gvrose8192/ovs-experimental/builds/644103253

Signed-off-by: Greg Rose <gvrose8192@gmail.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
---
 acinclude.m4                                 | 2 --
 datapath/linux/compat/include/linux/stddef.h | 7 -------
 datapath/linux/compat/include/linux/types.h  | 4 ----
 3 files changed, 13 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index c47a77ed6..1d16ff71c 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -835,8 +835,6 @@ AC_DEFUN([OVS_CHECK_LINUX_COMPAT], [
   OVS_GREP_IFELSE([$KSRC/include/linux/skbuff.h], [skb_nfct])
   OVS_GREP_IFELSE([$KSRC/include/linux/skbuff.h], [skb_put_zero])
 
-  OVS_GREP_IFELSE([$KSRC/include/linux/types.h], [bool],
-                  [OVS_DEFINE([HAVE_BOOL_TYPE])])
   OVS_GREP_IFELSE([$KSRC/include/linux/types.h], [__wsum],
                   [OVS_DEFINE([HAVE_CSUM_TYPES])])
   OVS_GREP_IFELSE([$KSRC/include/uapi/linux/types.h], [__wsum],
diff --git a/datapath/linux/compat/include/linux/stddef.h b/datapath/linux/compat/include/linux/stddef.h
index f2b7c319a..5b44c0dee 100644
--- a/datapath/linux/compat/include/linux/stddef.h
+++ b/datapath/linux/compat/include/linux/stddef.h
@@ -5,13 +5,6 @@
 
 #ifdef __KERNEL__
 
-#ifndef HAVE_BOOL_TYPE
-enum {
-	false   = 0,
-	true    = 1
-};
-#endif /* !HAVE_BOOL_TYPE */
-
 #ifndef offsetofend
 #define offsetofend(TYPE, MEMBER) \
 	(offsetof(TYPE, MEMBER)	+ sizeof(((TYPE *)0)->MEMBER))
diff --git a/datapath/linux/compat/include/linux/types.h b/datapath/linux/compat/include/linux/types.h
index b989d96c3..a58623e70 100644
--- a/datapath/linux/compat/include/linux/types.h
+++ b/datapath/linux/compat/include/linux/types.h
@@ -8,8 +8,4 @@ typedef __u16 __bitwise __sum16;
 typedef __u32 __bitwise __wsum;
 #endif
 
-#ifndef HAVE_BOOL_TYPE
-typedef _Bool bool;
-#endif /* !HAVE_BOOL_TYPE */
-
 #endif
-- 
2.17.1

